api-reference/ingestion/lease/post.mdx

```diff
@@ -0,0 +1,204 @@
+      Whether the lease is for a furnished or unfurnished property.
+    </ParamField>
+
+    <ParamField body="minimum_days_notice" type="integer" default="none">
+      Minimum days of notice that must be given to break the lease agreement.
+    </ParamField>
+
+    <ParamField body="deposit" type="float" default="none">
+      The deposit amount for the lease.
+    </ParamField>
+
+    <ParamField body="rent" type="float" default="none">
+      The rent amount for the lease.
+    </ParamField>
+  </Expandable>
+</ParamField>
+
+<ParamField body="tenants" type="[object]" default="none">
+  A list of tenants to be associated with the created or updated lease.
+  <Expandable title="tenant">
+      <ParamField body="email_address" type="string" default="none">
+        The email adddress of the tenant.
+      </ParamField>
+
+      <ParamField body="first_name" type="string" default="none">
+        The first name of the tenant.
+      </ParamField>
+
+      <ParamField body="last_name" type="string" default="none">
+        The last name of the tenant.
+      </ParamField>
+
+      <ParamField body="external_ref" type="string" default="none">
+        The external reference of the tenant.
+        
+        This is typically an identifier you may use for person records in your own system.
+      </ParamField>
+
+      <ParamField body="phone_number" type="string" default="none">
+        The phone number of the tenant.
+      </ParamField>
+  </Expandable>
+</ParamField>
+
+### Response
+
+<ParamField name="success" type="boolean">
+  Whether the API operation was successful or not.
+</ParamField>
+
+<ParamField name="lease" type="string">
+  The identifier for the modified or created lease.
+</ParamField>
+
+### Errors
+
+Listed below are common errors that may be returned by the endpoint, along with their corresponding status code.
+
+Status Code - 400
+```json Missing Authorization header
+{
+   "error": {
+    "type": "missing_authorization",
+    "message": "Your request does not include an 'Authorization' header with a bearer token for your account."
+  }
+}
+```
+
+Status Code - 401
+```json Invalid/Expired Authorization header bearer token value
+{
+   "error": {
+    "type": "invalid_authorization",
+    "message": "The bearer token you have provided in the 'Authorization' header is invalid."
+  }
+}
+```
+
+Status Code - 400
+
+Returned if more than one landlord is retrieved for the lease
+```json Could not retrieve a single landlord
+{
+   "error": {
+    "type": "multiple_landlord",
+    "message": "Could not retrieve a single landlord for the provided landlord_id."
+  }
+}
+```
+
+Status Code - 400
+
+Returned if a single tenant can not be retrieved for any of your tenant records
+```json Could not retrieve a single tenant
+{
+   "error": {
+    "type": "multiple_landlord",
+    "message": "Could not identify a single tenant record for one of the tenant objects you have provided."
+  }
+}
```

Comment:
Same as previous, should we have multiple 400s and if so can they be closer to each other?

Reply:
I see what you mean by closer :)

I grouped the first two errors together on the basis that they are related to authentication and they will show up for every endpoint in the API. I can switch them around so that the 400 comes after the 401, but to me it made more sense to have the authorization header missing one first and the auhtorization header invalid one second. Plus even if they are switched around, there could still be some other statuses between the authentication 400 and other 400s.

Alternatively I could further break down the grouping of errors so that the authentication errors are grouped under one sub-heading and the rest under other sub-headings. Do you think that's worth doing?

Reply:
I see, that actually makes sense. I'm fine with either, was just trying to understand why it was out of sequence. Thanks for explaining. Regarding a further breakdown of grouping, that's your call. I don't mind this structure knowing the reasoning behind it.